server/src/main/java/org/elasticsearch/common/util/AbstractBigByteArray.java

```diff
@@ -10,21 +10,31 @@
 
 import org.apache.lucene.util.ArrayUtil;
 import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.recycler.Recycler;
 
+import java.io.IOException;
 import java.util.Arrays;
 
 abstract class AbstractBigByteArray extends AbstractBigArray {
 
+    protected static final byte[] ZERO_PAGE = new byte[PageCacheRecycler.BYTE_PAGE_SIZE];
+
     protected byte[][] pages;
 
     protected AbstractBigByteArray(int pageSize, BigArrays bigArrays, boolean clearOnResize, long size) {
         super(pageSize, bigArrays, clearOnResize);
         this.size = size;
         pages = new byte[numPages(size)][];
-        for (int i = 0; i < pages.length; ++i) {
-            pages[i] = newBytePage(i);
+        Arrays.fill(pages, ZERO_PAGE);
+        assert assertZeroPageClean();
```

Comment:
Had some bugs with this, it's a little wasteful but having this in place lets me sleep better I think.